### ISSUE TITLE
binance-client - WARNING - [Sub] call re_connect suscribeuserdata.py

### ISSUE THREAD

**Issue by bladeride** [Binance-docs/Binance_Futures_python#63]
please try to correct the following error it is impossible to keep the connection stable.

https://prnt.sc/us72t8

**Comment by nikky78** [comment]
Any update about the issue? Warnings keep on popping and I miss information while reconnecting

**Comment by xilopaint** [comment]
Any news here?

**Comment by Alex1304** [comment]
Got the same issue, figured it's due to the fact that by default it considers the connection dead if no message is received in 60 seconds. But the user_data channel isn't verbose at all, so it is too easily triggered. A solution is to increase that timeout value when creating the SubscriptionClient (for example 1 hour):
```py
sub_client = SubscriptionClient(api_key='...', secret_key='...', receive_limit_ms=3600000)
```

**Comment by nagadomi** [comment]
This issue is especially problematic for UserData event with few updates. You can reproduce this issue with `example/user/subscribeuserdata.py`.
Once this happens, it seems to be impossible to recover.

Here's a dirty workaround I've done:
- Run in a separate process and reboot the process if problem is detected
- Post a useless order every 30 seconds :smile:

**Comment by Alex1304** [comment]
> This issue is especially problematic for UserData event with few updates. You can reproduce this issue with `example/user/subscribeuserdata.py`.
> Once this happens, it seems to be impossible to recover.
> 
> Here's a dirty workaround I've done:
> 
>     * Run in a separate process and reboot the process if problem is detected
> 
>     * Post a useless order every 30 seconds 😄

Did you see my comment right above? I explained the cause of the issue and the cleanest way to solve it ^^

**Comment by nagadomi** [comment]
@Alex1304 
I run a trading bot, so taking an hour to detect a problem is a big problem.
Also, if you extend the timeout to 1 hour, you should still have the same problem if there are no events for 1 hour.

**Comment by Alex1304** [comment]
Well imo assuming the connection is dead because no messages were received after x seconds isn't a reliable way to detect problems in the first place. Ideally there should be a heartbeat system where the client must send pings to the server at regular intervals, but not sure Binance implemented that.

In your case, I'd increase the timeout to something very high (like 99999999999999) and figure out another way to test whether the bot is down or still working.

**Comment by nagadomi** [comment]
@Alex1304 
I don't think this code itself is reliable... For example, this line calls `on_close` method,
https://github.com/Binance-docs/Binance_Futures_python/blob/a8e272d4708c8ba134ea7f5db5a87c52b55deaeb/binance_f/impl/websocketconnection.py#L34
but, `on_close` method is not defined in `WebsocketConnection` class.
So I want binance devs to take the fix of the issue seriously, not the user's workaround.
I myself can address this issue with other websocket implementations.
